### PR TITLE
Core: updated creature type flags enum

### PR DESCRIPTION
**Pull request by ForesterDev**
**Changes proposed**:
- Updated according https://trinitycore.atlassian.net/wiki/display/tc/creature_template#creature_template-type_flags

**Target branch(es)**: 335/6x

### CODE REVIEW

**Comment by Shauren** [comment]
You can add even more names, https://github.com/TrinityCore/WowPacketParser/blob/master/WowPacketParser/Enums/CreatureTypeFlag.cs

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 1]
```diff
@@ -2576,38 +2576,38 @@ enum CreatureFamily
...
+    CREATURE_TYPE_FLAG_DO_NOT_LOG_DEATH                     = 0x00000400,   // Death event will not show up in combat log
+    CREATURE_TYPE_FLAG_MOUNTED_COMBAT_ALLOWED               = 0x00000800,   // Creature can remain mounted when entering combat
+    CREATURE_TYPE_FLAG_CAN_ASSIST                           = 0x00001000,   // ? Can aid any player in combat if in range?
+    CREATURE_TYPE_FLAG_IS_PET_BAR_USED                      = 0x00002000,
+    CREATURE_TYPE_FLAG_MASK_UID                             = 0x00004000,
+    CREATURE_TYPE_FLAG_ENGINEERING_SKINNING_SKILL           = 0x00008000,   // Can be looted by engineer
+    CREATURE_TYPE_FLAG_EXOTIC_PET                           = 0x00010000,   // Can be tamed by hunter as exotic pet
+    CREATURE_TYPE_FLAG_USE_DEFAULT_COLLISION_BOX            = 0x00020000,   // Collision related. (always using default collision box?)
+    CREATURE_TYPE_FLAG_IS_SIEGE_WEAPON                      = 0x00040000,
+    CREATURE_TYPE_FLAG_DOES_NOT_COLLIDE_WITH_MISSILES       = 0x00080000,   // Projectiles can collide with this creature - interacts with TARGET_DEST_TRAJ
```
Comment (Treeston) [inline src/server/game/Miscellaneous/SharedDefines.h]: Projectiles _can_, or _cannot_? Comment says "can", but name says `DOES_NOT_COLLIDE`?

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 2]
```diff
@@ -2576,38 +2576,38 @@ enum CreatureFamily
...
     CREATURE_TYPEFLAGS_MASK_UID                             = 0x00004000,
-    CREATURE_TYPEFLAGS_ENGINEERLOOT                         = 0x00008000,   // Can be looted by engineer
-    CREATURE_TYPEFLAGS_EXOTIC                               = 0x00010000,   // Can be tamed by hunter as exotic pet
+    CREATURE_TYPEFLAGS_ENGINEERING_SKINNING_SKILL           = 0x00008000,   // Can be looted by engineer
+    CREATURE_TYPEFLAGS_EXOTIC_PET                           = 0x00010000,   // Can be tamed by hunter as exotic pet
     CREATURE_TYPEFLAGS_USE_DEFAULT_COLLISION_BOX            = 0x00020000,   // Collision related. (always using default collision box?)
     CREATURE_TYPEFLAGS_IS_SIEGE_WEAPON                      = 0x00040000,
-    CREATURE_TYPEFLAGS_PROJECTILE_COLLISION                 = 0x00080000,   // Projectiles can collide with this creature - interacts with TARGET_DEST_TRAJ
-    CREATURE_TYPEFLAGS_HIDE_NAMEPLATE                       = 0x00100000,
+    CREATURE_TYPEFLAGS_DOES_NOT_COLLIDE_WITH_MISSILES       = 0x00080000,   // Projectiles can collide with this creature - interacts with TARGET_DEST_TRAJ
```
Comment (joschiwald) [inline src/server/game/Miscellaneous/SharedDefines.h]: that sounds like the opposite

File: src/server/game/Miscellaneous/SharedDefines.h (hunk 1 above)
Comment (joschiwald) [inline src/server/game/Miscellaneous/SharedDefines.h]: implementation says "can"
creatures does not collide with missiles by default

Comment (Shauren) [inline src/server/game/Miscellaneous/SharedDefines.h]: yeah, name is wrong in wpp it seems (todo: open that old legendary 13850 cata debug build that was the source of this)

Comment (ForesterDev) [inline src/server/game/Miscellaneous/SharedDefines.h]: Should we rename this flag to CREATURE_TYPE_FLAG_CAN_COLLIDE_WITH_MISSILES according the implementation in TrinityCore, despite the name in WPP?

Comment (Shauren) [inline src/server/game/Miscellaneous/SharedDefines.h]: yes

**Comment by Aokromes** [comment]
Any reason to don't merge this?

**Pull request merged**